react-carousel/Carousel/src/Carousel.tsx

```diff
@@ -0,0 +1,47 @@
+import { Next } from './Next';
+import { Previous } from './Previous';
+import { Indicators } from './Indicators';
+import { Images } from './Images';
+import { useState, useEffect } from 'react';
+
+type Props = {
+  images: string[];
+};
+
+export function Carousel({ images }: Props) {
+  const [activeIndex, setActiveIndex] = useState(0);
+
+  useEffect(() => {
+    const intervalId = setTimeout(handleNext, 3000);
+    return () => {
+      clearInterval(intervalId);
+    };
+  });
```

Comment:
You don't have a dependency array here, so you'll get a new timeout interval on every render. But, you will also clear the previous one in your clean-up. So this might work. However, it's a bad practice to ever have a useEffect without a dependency array as it frequently leads to bugs.